### PR TITLE
added clip feature and integrated cut value ntuple producer (activate by building with -d flag)

### PR DESCRIPTION
**Pull request by bsathian**

### CODE REVIEW

File: SDL/Event.cu [hunk 1]
```diff
@@ -806,13 +806,13 @@ void SDL::Event::createSegmentsWithModuleMap()
     for (int i=0; i<nLowerModules; i++) {
       unsigned int innerLowerModuleIndex = modulesInGPU->lowerModuleIndices[i];
       unsigned int nConnectedModules = modulesInGPU->nConnectedModules[innerLowerModuleIndex];
-      unsigned int nInnerMDs = nMDs[innerLowerModuleIndex];
+      unsigned int nInnerMDs = nMDs[innerLowerModuleIndex] > N_MAX_MD_PER_MODULES ? N_MAX_MD_PER_MODULES : nMDs[innerLowerModuleIndex];
```
Comment (slava77) [inline SDL/Event.cu:809]:
```suggestion
      unsigned int nInnerMDs = std::min(nMDs[innerLowerModuleIndex],N_MAX_MD_PER_MODULES);
```
is shorter.
However, is there a `-1` missing?

Comment (slava77) [reply SDL/Event.cu:809]: >  However, is there a `-1` missing?

no, probably not

Comment (slava77) [reply SDL/Event.cu:809]: BTW, if the checks are done at write time, perhaps they are not really required at read time

**Comment by bsathian** [comment]
Efficiency plots here : http://uaf-10.t2.ucsd.edu/~bsathian/SDL/efficiencies_20210117/

I'm merging this so that I don't stack PRs on top of each other

**Pull request merged**